snuba/datasets/schema.py

```diff
@@ -8,46 +9,40 @@ def local_dataset_mode():
     return settings.DATASET_MODE == "local"
 
 
-class Schema(object):
+class Schema(ABC):
     """
-    Represents the full set of columns in a clickhouse table, this only contains
-    basic metadata for now.
+    A schema is an abstraction over a data model we can query.
+    It provides a set of columns and a where clause to build the query.
+    Concretely this can represent a table, a view or a group of
+    joined tables.
+    This level of abstraciton only provides read primitives.
+
+    Going on this will be abstract from Clickhouse details. As of
+    now this distinction does not exists to this provides CLickhouse
+    specific details.
     """
 
-    TEST_TABLE_PREFIX = "test_"
-
-    def __init__(self, local_table_name, dist_table_name, columns, migration_function=None):
+    def __init__(
+        self,
+        columns: ColumnSet,
+        migration_function: Optional[Callable[[str, Mapping[str, str]], Sequence[str]]]=None,
+    ) -> None:
         self.__columns = columns
+        self.__migration_function = migration_function if migration_function else lambda table, schema: []
 
-        self.__local_table_name = local_table_name
-        self.__dist_table_name = dist_table_name
-        self.__migration_function = migration_function if migration_function else lambda schema: []
-
-    def _make_test_table(self, table_name):
-        return table_name if not settings.TESTING else "%s%s" % (self.TEST_TABLE_PREFIX, table_name)
-
-    def get_local_table_name(self):
+    @abstractmethod
+    def get_where_clause(self) -> str:
         """
-        This returns the local table name for a distributed environment.
-        It is supposed to be used in DDL commands and for maintenance.
-        """
-        return self._make_test_table(self.__local_table_name)
+        Builds and returns the content of the where clause we need
+        to pass to Clickhouse to execute a query on this schema.
 
-    def get_table_name(self):
-        """
-        This represents the table we interact with to send queries to Clickhouse.
-        In distributed mode this will be a distributed table. In local mode it is a local table.
+        TODO: Once we have a Snuba Query abstraction (PR 456) this
```

Comment:
I tried to do that in this PR but, without the query structure it added complexity not needed to this PR.
It is still in one of the previous commits since I tried that way before this one:
https://github.com/getsentry/snuba/pull/467/files/96fe1a250c5a800eb7429a287387402db0d5df16

See the schema_storage file in that commit.